Modules/SaltStackConfig/SaltStackConfig.psm1

```diff
@@ -0,0 +1,304 @@
+    $webRequestJson = ConvertFrom-JSON $webRequest.Content
+    $global:DefaultSscConnection = New-Object psobject -property @{ "SscWebSession"=$ws; "SscServer"=$server; "ConnectionDetail"=$webRequestJson }
+    
+	# Return a few grains, like the Salt server & version; this will prove the connection worked & provide some context
+	(Get-SscMaster).ret.salt.grains | Select-Object Host, NodeName, SaltVersion, @{N='Authenticated';E={$global:DefaultSscConnection.ConnectionDetail.authenticated}}, 
+  @{N='AuthType';E={$global:DefaultSscConnection.ConnectionDetail.attributes.config_driver}}, @{N='AuthSource';E={$global:DefaultSscConnection.ConnectionDetail.attributes.config_name}}, 
+  @{N='UserName';E={$global:DefaultSscConnection.ConnectionDetail.attributes.username}}, @{N='Permissions';E={[string]::Join(', ', $global:DefaultSscConnection.ConnectionDetail.attributes.permissions)}}
+  } catch {
+    write-warning "Failure connecting to $server"
+  } # end try/catch block
+}
+
+Function Disconnect-SscServer { 
+<#
+  .NOTES
+  ===========================================================================
+   Created by:	Brian Wuchner
+   Date:		November 27, 2021
+   Blog:		www.enterpriseadmins.org
+   Twitter:		@bwuch
+  ===========================================================================
+  .SYNOPSIS
+    This function will clear the global variable used to connect to the vRealize Automation SaltStack Config API
+  .DESCRIPTION
+    This function clears a previously created cookie/header used to connect to SaltStack Config
+  .EXAMPLE
+    PS C:\> Disconnect-SscServer
+#>
+  if ($global:DefaultSscConnection) {
+    $global:DefaultSscConnection = $null 
+  } else {
+    write-warning "Not connected to any SaltStack Config servers."
+  } # end if
+}
+
+Function Get-SscData {
+<#
+  .NOTES
+  ===========================================================================
+   Created by:	Brian Wuchner
+   Date:		November 27, 2021
+   Blog:		www.enterpriseadmins.org
+   Twitter:		@bwuch
+  ===========================================================================
+  .SYNOPSIS
+    This function will pass resource/method/arguments to the vRealize Automation SaltStack Config API.
+    It depends on a global variable created by Connect-SscServer.
+  .DESCRIPTION
+    Use this function to call the SaltStack Config API.
+    Additional helper functions will call this function, this is where the majority of the logic will happen.
+  .EXAMPLE
+    PS C:\> Get-SscData -Resource 'minions' -Method 'get_minion_cache'
+#>
+  param(
+    [Parameter(Mandatory=$true)][string]$resource,
+    [Parameter(Mandatory=$true)][string]$method,
+    [System.Collections.Hashtable]$kwarg
+  )
+
+  if (!$global:DefaultSscConnection) {
+    write-warning "Not connected to any SaltStack Config servers."
+    return;
+  } # end if
+
+  if (!$kwarg) {
+    $body = "{`"resource`": `"$resource`", `"method`": `"$method`"}"

```

Comment:
It will be easier to read if you make the $body variable a hash table and then use ConvertTo-Json $body to pass it to Invoke-WebRequest

Reply:
Agreed, I'll get this fixed in the next commit.